go/arrow/csv/csv_test.go

```diff
@@ -173,35 +169,25 @@ func TestCSVReader(t *testing.T) {
 		},
 		nil,
 	)
-	r := csv.NewReader(bytes.NewReader(raw), schema,
-		csv.WithAllocator(mem),
-		csv.WithComment('#'), csv.WithComma(';'),
-	)
-	defer r.Release()
-
-	r.Retain()
-	r.Release()
-
-	if got, want := r.Schema(), schema; !got.Equal(want) {
-		t.Fatalf("invalid schema: got=%v, want=%v", got, want)
-	}
-
-	out := new(bytes.Buffer)
-
-	n := 0
-	for r.Next() {
-		rec := r.Record()
-		for i, col := range rec.Columns() {
-			fmt.Fprintf(out, "rec[%d][%q]: %v\n", i, rec.ColumnName(i), col)
-		}
-		n++
-	}
-
-	if got, want := n, 2; got != want {
-		t.Fatalf("invalid number of rows: got=%d, want=%d", got, want)
-	}
-
-	want := `rec[0]["bool"]: [true]
+	for _, tc := range []struct {
+		name    string
+		csv     io.Reader
+		opts    []csv.Option
+		records int
+		want    string
+	}{
+		{
+			name: "including various values which doesn't contain null values",
```

Comment:
that's a bit of a mouthful.
also, it makes it harder to selectively select or disable a given set of sub-tests.

could we come up with a shorter yet descriptive name?
perhaps something along the lines of `without-null-values`?